File: android/src/main/java/com/margelo/nitro/camera/facedetector/HybridFaceDetectorFactory.kt
package com.margelo.nitro.camera.facedetector

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip
import com.margelo.nitro.camera.HybridCameraOutputSpec

@DoNotStrip
@Keep
class HybridFaceDetectorFactory : HybridFaceDetectorFactorySpec() {
  @DoNotStrip
  @Keep
  override fun createFaceDetector(options: FaceDetectorOptions): HybridFaceDetectorSpec {
    return HybridFaceDetector(options)
  }

  @DoNotStrip
  @Keep
  override fun createFaceDetectorOutput(options: FaceDetectorOutputOptions): HybridCameraOutputSpec {
    return HybridFaceDetectorOutput(options)
  }
}

File: ios/HybridFaceDetectorFactory.swift
import NitroModules
import VisionCamera

class HybridFaceDetectorFactory: HybridFaceDetectorFactorySpec {
  func createFaceDetector(options: FaceDetectorOptions) throws -> any HybridFaceDetectorSpec {
    return HybridFaceDetector(options: options)
  }

  func createFaceDetectorOutput(options: FaceDetectorOutputOptions) throws
    -> any HybridCameraOutputSpec
  {
    return HybridFaceDetectorOutput(options: options)
  }
}
